numpy/lib/function_base.py

```diff
@@ -1711,12 +1711,17 @@ def diff(a, n=1, axis=-1):
         The number of times values are differenced.
     axis : int, optional
         The axis along which the difference is taken, default is the last axis.
+    to_begin : 1d array_like
```

Comment:
Add a comment here about the required shape please. This should work for arbitrary-dimensional `a`.